instr.ml

```diff
@@ -322,3 +322,82 @@ let checkpoint_label pc =
 
 let independent instr exp =
   VarSet.is_empty (VarSet.inter (changed_vars instr) (expr_vars exp))
+
+class mapper = object (m)
```

Comment:
I found the naming a bit confusing.

1. I would call the thing a *Map* instead of a *Mapper*
2. If you really want *Map* as a type then it should be an interface (not sure if they exist in ocaml, no need to actually do this, since this is a simple case)
3. This class here is the *IdentityMap*